filters/include/pcl/filters/impl/uniform_sampling.hpp

```diff
@@ -111,9 +111,11 @@ pcl::UniformSampling<PointT>::applyFilter (PointCloud &output)
       continue;
     }
 
+	// Compute the voxel center
+	Eigen::Vector4f voxel_center((ijk[0] + 0.5) * leaf_size_[0], (ijk[1] + 0.5) * leaf_size_[1], (ijk[2] + 0.5) * leaf_size_[2], 1);
```

Comment:
I don't know if you're supposed to use the voxel center, but if you are:

```cpp
ijk[3] = 0.5;
const Eigen::Vector4f voxel_center = (ijk + 0.5) * leaf_size_;
```

PS: Does voxel_center[3] have to be 1 or 0?

Reply:
Yours is the easier.
In theory, voxel_center[3] should be equal to 1 because of the following (data[3] = 1.0f;). 
https://github.com/PointCloudLibrary/pcl/blob/3cb0cf4a2120a16c8527abcb366b76372fcf54ba/common/include/pcl/impl/point_types.hpp#L309
But if voxel_center[3] is equal to 0, you should also get the right results .

Regards,

Reply:
I tested the code you suggested, but it was compiled incorrectly and displayed scalar cannot be added to the Vector4i type. I modify the code, and commit it, please check it. @kunaltyagi